tests/dataset/measurement/test_measurement_context_manager.py

```diff
@@ -730,6 +734,15 @@ def test_datasaver_scalars(
         with pytest.raises(ValueError):
             datasaver.add_result((DMM.v1, 0))
 
+    ds = datasaver.dataset
+    assert isinstance(ds, DataSet)
+    assert isinstance(ds.description.interdeps, FrozenInterDependencies_)
```

Comment:
This is not the most intuitive however to benefit from the speedup we need to freeze the Interdependencies_ that is passed to the dataset while we measure since that is used by add_result and friends. 
We cannot currently unfreeze this at the end of the measurement since the rundescripter that holds the interdeps has been marked as frozen and no longer can be modified.

Reply:
Pushed a change to allow modification on datasaver exit. Not the nicest code but I think its the best solution for now